packages/filesystem/src/browser/filesystem-frontend-contribution.ts

```diff
@@ -134,6 +148,22 @@ export class FileSystemFrontendContribution implements FrontendApplicationContri
                 }
             }
         });
+        commands.registerCommand(CommonCommands.NEW_FILE, {
+            execute: async (...args: unknown[]) => {
+                let newFileName = undefined;
+                if (args !== undefined && typeof args[1] === 'string') {
+                    newFileName = args[1];
+                }
+                const title = nls.localizeByDefault('Create File');
+                const props: SaveFileDialogProps = { title, saveLabel: title, inputValue: newFileName };
+                const selectedFile = this.getSelection(...args)?.fileStat ?? await this.getFolderFromArgs(...args);
```

Comment:
I really like that we are trying to give our best to provide a good parent directory. However, I'm a bit concerned about the order. I believe that when a directory is explicitly given as argument to the command that it should have precedence. So maybe we should inject the `UserWorkingDirectoryProvider` here as well and then do have the following order: explicitly given argument > selection > user directory fallback.

What do you think?

Reply:
As the selection is checked inside the `UserWorkingDirectoryProvider` the order is:  
explicitly given argument  > user directory fallback.

Makes it simpler, I like it.